src/pu-emmc.c

```diff
@@ -401,54 +404,61 @@ pu_emmc_write_data(PuFlash *flash,
             return FALSE;
     }
 
-    if (self->emmc_boot_partitions) {
-        PuEmmcInput *input = self->emmc_boot_partitions->input;
-        g_autofree gchar *path = NULL;
+    if (self->mmc_controls) {
+        PuEmmcBootPartitions *boot_partitions = NULL;
 
-        path = pu_path_from_filename(input->filename, prefix, error);
-        if (path == NULL) {
-            g_prefix_error(error, "Failed parsing input filename for eMMC boot partition: ");
-            return FALSE;
-        }
+        boot_partitions = self->mmc_controls->boot_partitions;
+        if (boot_partitions && pu_has_bootpart(self->device->path)) {
+            PuEmmcInput *input = boot_partitions->input;
 
-        if (g_str_equal(path, "")) {
-            g_set_error(error, PU_ERROR, PU_ERROR_FLASH_DATA,
-                        "No input specified for eMMC boot partition");
-            return FALSE;
-        }
-
-        if (!g_str_equal(input->md5sum, "") && !skip_checksums) {
-            g_debug("Checking MD5 sum of input file '%s'", path);
-            if (!pu_checksum_verify_file(path, input->md5sum,
-                                         G_CHECKSUM_MD5, error))
+            if (!pu_bootpart_enable(self->device->path, boot_partitions->enable, error))
```

Comment:
The second argument for `pu_bootpart_enable()` shoud be `guint` instead of `gboolean`. It is not only used to enable and disable the bootpartition, but also to select the bootpartition.